cmd/install/install.go

```diff
@@ -0,0 +1,228 @@
+		}
+		_, err2 := exec.Command("chmod", "+x", kubectlPath).Output()
+		if err2 != nil {
+			fmt.Println(" → Error giving access to kubectl")
+			return err2
+		}
+
+	case "windows":
+		err1 := downloadKubectlLatest(osPlatform)
+		if err1 != nil {
+			fmt.Println(" → Error installing the latest kubectl version")
+		}
+		fmt.Println(" → Add kubectl binary to your windows path")
+
+	case "linux":
+		err1 := downloadKubectlLatest(osPlatform)
+		if err1 != nil {
+			fmt.Println(" → Error installing the latest kubectl version")
+		}
+		_, err2 := exec.Command("chmod", "+x", kubectlPath).Output()
+		if err2 != nil {
+			fmt.Println(" → Error giving access to kubectl")
+			return err2
+		}
+	}
+	return nil
+}
+
+// Downloads the latest version of kubectl
+func downloadKubectlLatest(osPlatform string) error {
+	latestVersion := "https://storage.googleapis.com/kubernetes-release/release/stable.txt"
+	ver, err := exec.Command("curl", "-sL", latestVersion).Output()
+	if err != nil {
+		fmt.Println(" → Error fetching latest kubectl version")
+		return err
+	}
+	var downloadKubectl string
+	if osPlatform == "windows" {
+		downloadKubectl = "https://storage.googleapis.com/kubernetes-release/release/" + string(ver) + "/bin/" + osPlatform + "/amd64/kubectl.exe"
+	} else {
+		downloadKubectl = "https://storage.googleapis.com/kubernetes-release/release/" + string(ver) + "/bin/" + osPlatform + "/amd64/kubectl"
+	}
+	_, err1 := exec.Command("curl", "-o", kubectlPath, "-LO", downloadKubectl).Output()
+	if err1 != nil {
+		fmt.Println(" → Error downloading latest kubectl version")
+		return err1
+	}
+	return nil
+}
+
+// Check the platform and on the basis of that install argocd
+func installArgoCDPlatform() error {
+	osPlatform := runtime.GOOS
+	fmt.Println(" → Installing argocd")
+	switch osPlatform {
+	case "darwin":
+		err1 := downloadArgoCDLatest(osPlatform)
+		if err1 != nil {
+			fmt.Println(" → Error installing the latest argocd version")
+		}
+		_, err2 := exec.Command("chmod", "+x", argocdPath).Output()
+		if err2 != nil {
+			fmt.Println(" → Error giving access to argocd")
+			return err2
+		}
+
+	case "windows":
+		err1 := downloadArgoCDLatest(osPlatform)
+		if err1 != nil {
+			fmt.Println(" → Error installing the latest argocd version")
+		}
+		fmt.Println(" → Add argocd binary to your windows path")
+
+	case "linux":
+		err1 := downloadArgoCDLatest(osPlatform)
+		if err1 != nil {
+			fmt.Println(" → Error installing the latest argocd version")
+		}
+		_, err2 := exec.Command("chmod", "+x", argocdPath).Output()
+		if err2 != nil {
+			fmt.Println(" → Error giving access to argocd")
+			return err2
+		}
+	}
+	return nil
+}
+
+// Downloads the latest version of argocd
+func downloadArgoCDLatest(osPlatform string) error {
+	var downloadArgoCD string
+	if osPlatform == "windows" {
+		downloadArgoCD = "https://github.com/argoproj/argo-cd/releases/download/v2.2.12/argocd-" + osPlatform + "-amd64.exe"
```

Comment:
Make this a constant/ global for like ARGOCD_VERSION since it's going to change and CLIs of ArgoCD 2.2 is not compatible with ArgoCD 2.4

Reply:
I will add a argocd_version variable